src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageItemViewModel.cs

```diff
@@ -494,14 +494,38 @@ public async Task<IReadOnlyCollection<VersionInfoContextInfo>> GetVersionsAsync(
         {
             var identity = new PackageIdentity(Id, Version);
             var isTransitive = PackageLevel == PackageLevel.Transitive;
-            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, _cancellationTokenSource.Token);
+
+            CancellationToken cancellationToken;
+            try
+            {
+                cancellationToken = _cancellationTokenSource.Token;
+            }
+            catch (ObjectDisposedException)
+            {
+                // If the CancellationTokenSource is disposed, return empty collection
+                return Array.Empty<VersionInfoContextInfo>();
+            }
+
+            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, cancellationToken);
         }
 
         public async Task<IReadOnlyCollection<VersionInfoContextInfo>> GetVersionsAsync(IEnumerable<IProjectContextInfo> projects)
         {
             var identity = new PackageIdentity(Id, Version);
             var isTransitive = PackageLevel == PackageLevel.Transitive;
-            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, projects, _cancellationTokenSource.Token);
+
+            CancellationToken cancellationToken;
+            try
+            {
+                cancellationToken = _cancellationTokenSource.Token;
+            }
+            catch (ObjectDisposedException)
+            {
+                // If the CancellationTokenSource is disposed, return empty collection
+                return Array.Empty<VersionInfoContextInfo>();
+            }
+
+            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, projects, cancellationToken);
```

Comment:
So this is one of those where there's a race condition correct? 
What if it gets disposed of when this operation is happening? 
Is that possible? 
It'd still throw and we'd log the exception.